accepted/0000-no-package-json-_fields.md

```diff
@@ -0,0 +1,274 @@
+# Do Not Put Meta Information In `_`-Prefixed `package.json` Fields
+
+## Summary
+
+Currently, npm stashes various information in the `package.json` files that
+it creates.
+
+We will stop putting metadata in `package.json` files in `node_modules`.
+The fields that serve a purpose will be relocated to
+`node_modules/.arborist-resolutions.json`.  When this metadata does not
+exist, npm will make a best effort to gather metadata from
+`package-lock.json` or `yarn.lock`.
+
+## Motivation
+
+These fields are only relevant to npm, and as of version 6, npm relies on
```

Comment:
This is incorrect; the `requiredBy` field has been *incredibly* invaluable when debugging dep graph issues. Could we find a way to have a simple answer to "what package(s) required package x"?

Perhaps a v6 minor that adds a CLI command that can show this info; then you could change the behind the scenes source of that info without breaking anyone?

Reply:
Ack!  I forgot `_requiredBy`, yeah, that's one of the most interesting ones.  I'll get to it  tomorrow.

> Could we find a way to have a simple answer to "what package(s) required package x"?

How would that differ from `npm ls x` today?

And don't worry, there is no way we're gonna land this before v7, it's definitely a breaking change :)

Reply:
I guess you're suggesting something where you can specify a single instance of a package folder in the tree, and it'd show all the cases where a dependency deduped to it?

Reply:
That sounds like what I'm asking for - essentially, I want the answer to the question "why is this dependency here".

Reply:
Updated this PR to include `_requiredBy` and note the intention to create  `npm ls <location>`, where `<location>` can be either a logical path, like `/tap/react` or a physical path like `./node_modules/tap/node_modules/react`.